### PR TITLE
fix(DatePicker): error not showing on date reset by button

### LINKED ISSUE #8905: Bug - DatePicker - Error stays when resetting the date
The error does not disappear when resetting the invalid value of the DatePicker to a blank one by not manually deleting it.

**Steps to reproduce:**
https://codesandbox.io/s/determined-banzai-h2876p?file=/index.tsx
1) Write some invalid date into the text input of the DateTime component (e.g. "a")
2) Click the "Clear date" button
=> The error message and the error icon do not disappear as they should

I am using @patternfly/react-core@4.271.2.

### PR DESCRIPTION
**Pull request by adamviktora**
**What**: Closes #8905

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]

File: packages/react-core/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -40,8 +40,12 @@ export interface DatePickerProps
   inputProps?: TextInputProps;
   /** Flag indicating the date picker is disabled. */
   isDisabled?: boolean;
+  /** Flag indicating the date picker is required. */
+  isRequired?: boolean;
```
Comment (tlabaj) [inline packages/react-core/src/components/DatePicker/DatePicker.tsx:44]: Could you rebase.  Another PR went in to fix a different issue and it added an API for `isRequred` and `emptyDateText`

**Review by tlabaj — CHANGES_REQUESTED** [review]

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -50,6 +50,8 @@ export interface DatePickerProps
   isDisabled?: boolean;
   /** Error message to display when the text input contains a non-empty value in an invalid format. */
   invalidFormatText?: string;
+  /** Error message to display when the text input is empty and the isRequired prop is also passed in. */
```
Comment (tlabaj) [inline packages/react-core/src/components/DatePicker/DatePicker.tsx:53]: you should nit need this.  you cna use the one in the object on line 20.

**Review by tlabaj — CHANGES_REQUESTED** [review]
one small comment

File: packages/react-core/src/components/DatePicker/DatePicker.tsx (hunk 2 above)
Comment (adamviktora) [reply packages/react-core/src/components/DatePicker/DatePicker.tsx]: sure, I forgot to delete that prop

**Review by tlabaj — APPROVED** [review]
LGTM

**Pull request merged**